cmd/cli/main.go

```diff
@@ -41,6 +42,7 @@ func main() {
 	if err := env.Parse(&agentGRPCConfig, env.Options{Prefix: envPrefixAgentGRPC}); err != nil {
 		logger.Fatal(fmt.Sprintf("failed to load %s gRPC client configuration : %s", svcName, err))
 	}
+	agentGRPCConfig.URL = defURL
```

Comment:
This shouldn't be the case. You are effectively writing the deafultURL to a fixed values after loading the env configs. Have a look at https://github.com/caarlos0/env?tab=readme-ov-file#defaults-from-code. You can specify the default values when loading env vars